src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1693,41 +1695,61 @@ ChildNodeList(MultiNode *multiNode)
 
 /*
  * UniqueJobId allocates and returns a unique jobId for the job to be executed.
- * This allocation occurs both in shared memory and in write ahead logs; writing
- * to logs avoids the risk of having jobId collisions.
  *
- * Please note that the jobId sequence wraps around after 2^32 integers. This
- * leaves the upper 32-bits to slave nodes and their jobs.
+ * The resulting job ID is built up as:
+ * <16-bit group ID><24-bit process ID><1-bit secondary flag><23-bit local counter>
+ *
+ * When citus.enable_unique_job_ids is off then only the local counter is
+ * included to get repeatable results.
```

Comment:
Wouldn't it be better to only change the *printing* of the result, rather than the actual behaviour?